### DISCUSSION TITLE
Can't work out the Typescript syntax for an empty array

### DISCUSSION THREAD

**Discussion by andrewpatto** [edgedb/edgedb-js#357] [Q&A] [answered]
I can make a string array `type` with

```
e.array(e.str)
```

But I can't work out the syntax to make an empty `instance` of a string array?

```
e.array()

e.array([])
```
both fail with compile errors (so I can't even get an object to attempt to cast to my type - as I would do for empty sets)

**Comment by jaclarke** [comment, marked as answer]
You need to use `e.literal` like this:
```
e.literal(e.array(e.str), []);
```
Docs: https://www.edgedb.com/docs/clients/01_js/types#custom-literals

**Comment by andrewpatto** [comment]
Thanks! That "custom literals" info is on the Types page - not the Literals page of the documentation (where I was looking). Possibly some cross reference or something? The discussion for how to make Empty sets is on the Literals page, so I stopped looking there when I couldn't find anything about arrays.

Thanks for an awesome db - am really enjoying using edgedb.

**Comment by andrewpatto** [comment]
In practice on execution I was then getting an 'indeterminate type' message - so I had to add a cast as well (as per the empty set example)

```
e.cast(e.array(e.str), e.literal(e.array(e.str), []));
```
